test/base/OpaqueId.test.cc

```diff
@@ -54,3 +55,21 @@ TEST(OpaqueIdTest, operations)
     EXPECT_EQ(10, Id_t{22} - Id_t{12});
     EXPECT_TRUE(Id_t{22} < Id_t{23});
 }
+
+TEST(OpaqueIdTest, multi_int)
+{
+    using SId8   = OpaqueId<TestInstantiator, std::int_least8_t>;
+    using UId8   = OpaqueId<TestInstantiator, std::uint_least8_t>;
+    using Uint32 = std::uint_least32_t;
+
+    // Unassigned is always out-of-range
+    EXPECT_FALSE(SId8{} < 0);
+    EXPECT_FALSE(SId8{} < Uint32(0xffffffffu));
```

Comment:
Not completely necessary (I know you love you some hexadecimal) but for the particle physicists on our team, it may be useful to just put in a comment:
```cpp
// 0xffffffff is maximum of a 32-bit unsigned integer, 2^32 - 1 = 4294967295
```
Like I said, not strictly necessary

Reply:
Ha ha, you caught me -- I'll just replace with `std::numeric_limits<uint32>::max()`  which  is what I should have used in the first place.

`0xdeadbeef`

Reply:
BTW I counted the `f`'s this time 🤣 and ironically I was one short the first time I typed it... bad habit to break I guess!